### PR TITLE
Fix(eos_validate_state): Update error message when a device does not have httpapi _sub_plugin

### LINKED ISSUE #3882: Fix(eos_validate_state): Better error message needed when a device does not have `httpapi` _subplugin
### Issue Summary

Impact AVD 4.7.x and devel

When running `eos_validate_state` with ANTA preview mode, if the devices are not configured with httpapi the following traceback can be seen

```
TASK [arista.avd.eos_validate_state : Run eos_validate_state_runner leveraging ANTA] ********************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'Connection' object has no attribute '_sub_plugin'
fatal: [mpls-rr1]: FAILED! => changed=false 
  msg: 'Error during plugin execution: ''Connection'' object has no attribute ''_sub_plugin'''
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'Connection' object has no attribute '_sub_plugin'
fatal: [mpls-site209-m9]: FAILED! => changed=false
```

This is expected as the code is trying to access a subplugin info that is not present on the device.

However AVD should output a better error message to explain what is going on.

### Which component(s) of AVD impacted

eos_validate_state

### How do you run AVD ?

None

### Steps to reproduce

```shell
cf description
```

### Relevant log output

```shell
cf description
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Better error message needed when a device does not have httpapi _sub_plugin

## Related Issue(s)

Fixes #3882 

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes
Added the If condition to check _sub_plugin is there or not.

## How to test
To reproduce issue we have to comment out ansible_connection vars.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py [hunk 1]
```diff
@@ -84,6 +84,9 @@ def __init__(self, name: str, connection: ConnectionBase, tags: list | None = No
 
         super().__init__(name, tags, disable_cache=False)
         self.check_mode = check_mode
+        # Check the ansible connection is defined
+        if not self.check_mode and not hasattr(connection, "_sub_plugin"):
+            raise AristaAvdError(message="Unable to determine ansible connection. Please manually configure ansible_connection as `httpapi` for this host.")
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py]: May be we can make it like:

```suggestion
        # Check the ansible connection is defined
        try:
            if self.check_mode or (plugin_name := connection._sub_plugin.get("name")) == ANSIBLE_EOS_PLUGIN_NAME:
                self._connection = connection
            else:
                raise AristaAvdError(message=f"The provided Ansible connection does not use EOS HttpApi plugin: {plugin_name}")
        except AttributeError:
            raise AristaAvdError(message="AVD could not determine ansible connection. Please configure ansible_connection as `httpapi` for this host.")
```

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py]: updated

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py [hunk 2]
```diff
@@ -84,11 +84,15 @@ def __init__(self, name: str, connection: ConnectionBase, tags: list | None = No
...
-            raise AristaAvdError(message=f"The provided Ansible connection does not use EOS HttpApi plugin: {plugin_name}")
+
+        try:
+            # In check_mode we don't care that we cannot connect to the device
+            if self.check_mode or (plugin_name := connection._sub_plugin.get("name")) == ANSIBLE_EOS_PLUGIN_NAME:
+                self._connection = connection
+            else:
+                raise AristaAvdError(message=f"The provided Ansible connection does not use EOS HttpApi plugin: {plugin_name}")
+        except AttributeError as err:
+            raise AristaAvdError(message="AVD could not determine ansible connection. Please configure ansible_connection as `httpapi` for this host.") from err
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py]:
```suggestion
            raise AristaAvdError(message="AVD could not determine the Ansible connection plugin used. Please ensure that the 'ansible_network_os' and 'ansible_connection' variables are set to 'eos' and 'httpapi' respectively for this host.") from err
```

**Review by carl-baillargeon — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py (hunk 2 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py]: Fixed.
If ansible_network_os is not defined and ansible_connection defined as httpapi then we face below error
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'Connection' object has no attribute 'nonetype'
fatal: [leaf2-dc1]: FAILED! => {"msg": "Unexpected failure during module execution: 'Connection' object has no attribute 'nonetype'", "stdout": ""}
```
and this is from netcommon collection
```
File "/home/maheshkumar/Projecct/ansible-avd/ansible_collections/ansible/netcommon/plugins/plugin_utils/connection_base.py", line 50, in __getattr__
    raise AttributeError(
AttributeError: 'Connection' object has no attribute 'nonetype'
fatal: [leaf2-dc1]: FAILED! => {
    "msg": "Unexpected failure during module execution: 'Connection' object has no attribute 'nonetype'",
    "stdout": ""
}
```
cc: @gmuloc

Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py]: Lets make all this nice maybe outside of the try except

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py]: done

**Review by gmuloc — APPROVED** [review]

**Review by carl-baillargeon — APPROVED** [review]
LGTM

**Pull request merged**